### ISSUE TITLE
[BUG] DNS spamming

### ISSUE THREAD

**Issue by BirdInFire** [faldez/tanoshi#295]
When tanoshi running with manganato, it research the dns of readmanganato.com two time by second even when not used.
Is there any way to reduce this DNS spamming ?

**Comment by faldez** [comment]
hmm it shouldn't do that, it only make request when explicitly told to or on periodic checks. are you sure it's from tanoshi?

**Comment by BirdInFire** [comment]
> hmm it shouldn't do that, it only make request when explicitly told to or on periodic checks. are you sure it's from tanoshi?

yes, actually i don't really know what's generate it, i do not really know what cause it but, after 5 or 6 (kill from portainer) and restart it stop and after some time it restart.
![image](https://user-images.githubusercontent.com/42673166/164984670-c935402a-995a-43ed-8f68-acda4a8fb691.png)
the main "simptom" is pictures who miss during a reading (with all the chapter loaded at once).

i have reduced the problem by adding a ridiculous cache on this domain, but i wonder where it come from :/

**Comment by faldez** [comment]
does it only happen for manganato? how about the other source?

**Comment by BirdInFire** [comment]
> does it only happen for manganato? how about the other source?

i will add another sources, and see if it happen in the same time come back when the bug restart

**Comment by BirdInFire** [comment]
> does it only happen for manganato? how about the other source?

is it possible that this problem come from the update check (each update check generate one dns request, and with ipv6 check it generate 2 hundred dns request) ?, but i have an idea to fix this problem (even if it's only a rare bug) add a dns cache of 2 hour into the docker image (or docker compose) file.

like that if it's a bug caused by the search it will worsen each time we add a source, so, a cache can absorb these excessive request.

**Comment by BirdInFire** [comment]
> does it only happen for manganato? how about the other source?

so there it is, after some research (and reading all the logs, it totally appen when the worker search after update, and there is exactly one dns request (one A and one AAAA) by series.
it reach the cache of local host (so it does not flood the upstream dns) but those who doesn't have a cache or (like me log the 6 last hour to found tracker or ads) it totally flood this logs.

Not a major problem,, since it happen one per hour but can be usefull to warn into a doc to actually configure a correct dns cache, (for those who use dnscrypt with one key by request (witch can really put the cpu on it's knee on lower hardware)).

**Comment by faldez** [comment]
Sorry for the late response, is this specific to tanoshi?

**Comment by BirdInFire** [comment]
the best way to replicate is to install adguard home, point the server dns to himself, and use doh on upstream (to be sure to capture all of them)
and you will see reading / updating create an absurd amound of DNS request.
